code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/mushroom
+
+/obj/item/food/pizza/mushroom/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/mushroom, 6, 30)
+
+/obj/item/food/pizza/mushroom/raw
+	name = "raw mushroom pizza"
+	icon_state = "mushroompizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/mushroom/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/mushroom, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice/mushroom
+	name = "mushroom pizza slice"
+	desc = "Maybe it is the last slice of pizza in your life."
+	icon_state = "mushroompizzaslice"
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizza/vegetable
+	name = "vegetable pizza"
+	desc = "No one of Tomatos Sapiens were harmed during making this pizza."
+	icon_state = "vegetablepizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/medicine/oculine = 12, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 2, "cheese" = 1, "carrot" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/vegetable
+
+/obj/item/food/pizza/vegetable/Initialize()
+	. = ..()
+	AddElement(/datum/element/processable, TOOL_KNIFE, /obj/item/food/pizzaslice/vegetable, 6, 30)
+
+/obj/item/food/pizzaslice/vegetable
+	name = "vegetable pizza slice"
+	desc = "A slice of the most green pizza of all pizzas not containing green ingredients."
+	icon_state = "vegetablepizzaslice"
+	tastes = list("crust" = 1, "tomato" = 2, "cheese" = 1, "carrot" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/food/pizza/vegetable/raw
+	name = "raw vegetable pizza"
+	icon_state = "vegetablepizza_raw"
+	foodtypes =  GRAIN | VEGETABLES | DAIRY | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/vegetable/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza/vegetable, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizza/donkpocket
+	name = "donkpocket pizza"
+	desc = "Who thought this would be a good idea?"
+	icon_state = "donkpocketpizza"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 20, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/medicine/omnizine = 10, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1, "laziness" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY | MEAT | JUNKFOOD
+	slice_type = /obj/item/food/pizzaslice/donkpocket
+
+/obj/item/food/pizza/donkpocket/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.